web-service/helpers.py

```diff
@@ -321,23 +322,24 @@ def StoreUrl(siteInfo, url, content, encoding):
         icon = urljoin(url, '/favicon.ico')
     # make sure the icon exists
     try:
-        result = urlfetch.fetch(icon, method = 'HEAD')
+        result = urlfetch.fetch(icon, validate_certificate=True)
         if result.status_code != 200:
             icon = None
         else:
             contentType = result.headers['Content-Type']
-            if contentType is None:
```

Comment:
It will trigger an exception but you're right, it's better to be explicit. 
Amended patch to follow.